widgetssdk/src/main/java/com/glia/widgets/helper/ActivityWatcher.kt

```diff
@@ -0,0 +1,49 @@
+package com.glia.widgets.helper
+
+import android.app.Activity
+import android.app.Application
+import android.os.Bundle
+import androidx.annotation.VisibleForTesting
+import com.glia.widgets.callvisualizer.controller.CallVisualizerController
+import java.lang.ref.WeakReference
+
+class ActivityWatcher(
+    private val app: Application,
+    private val callVisualizerController: CallVisualizerController
+) :
+    Application.ActivityLifecycleCallbacks {
+
+    private var _resumedActivity: WeakReference<Activity?> = WeakReference(null)
+
+    /**
+     * Returns last activity that called [Activity.onResume], but didn't call [Activity.onPause] yet
+     * @return Currently resumed activity.
+     */
+    @VisibleForTesting
+    val resumedActivity: Activity?
+        get() = _resumedActivity.get()
+
+    fun init() {
+        app.registerActivityLifecycleCallbacks(this)
+    }
+
+    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {}
+    override fun onActivityDestroyed(activity: Activity) {}
+
+
+    override fun onActivityResumed(activity: Activity) {
+        _resumedActivity = WeakReference(activity)
```

Comment:
does it make sense here to keep the activity instance in WeakReference? It is immediately passed here to another class, which is directly dealing with the activity instance, I think it would be better to get it from WeakReference directly before use or even maybe it makes sense to make CallVisualizer itself a kind of Activitywatcher(extend from Application.ActivityLifecycleCallbacks), this way CallVisualizer can itself wrap the context with WeakReference and use it from there